js/app.jsx

```diff
@@ -297,101 +416,94 @@ var CaseverListItem = React.createClass({
   }
 });
 
+var SearchTableHeadConstTh = React.createClass({
+  render: function(){
+    return(
+      <th>{this.props.text}</th>
+    )
+  }
+})
 
-var CaseverList = React.createClass({
+var SearchTableHeadTh = React.createClass({
+  handleSort: function(){
+    var newOrder = "";
+    if (this.state.order == "") {
+      newOrder = "-";
+    }
+    this.setState({"order": newOrder})
+    this.props.handleAddFilter(' orderby:' + newOrder + this.props.filter, / orderby:[-\w]+/)
+  },
+  getInitialState: function(){
```

Comment:
The call to `this.props.handleAddFilter` is correct, and should be kept. But the arrow icon can be control by `this.props` instead of `this.state`, `this.props` is immutable so it's safer. This `props` can be passed down from higher level components.